Libs/MRML/DisplayableManager/vtkMRMLInteractionWidgetRepresentation.cxx

```diff
@@ -183,26 +183,29 @@ vtkMRMLInteractionWidgetRepresentation::CanInteractWithCircleHandle(
   displayPosition3[1] = static_cast<double>(displayPosition[1]);
 
   double handleDisplayPos[4] = { 0.0, 0.0, 0.0, 1.0 };
+  double* handleWorldPos = handleInfo.PositionWorld;
   vtkMRMLSliceNode* sliceNode = this->GetSliceNode();
   if (sliceNode)
   {
     vtkNew<vtkMatrix4x4> rasToxyMatrix;
     vtkMatrix4x4::Invert(sliceNode->GetXYToRAS(), rasToxyMatrix);
 
-    double* handleWorldPos = handleInfo.PositionWorld;
     rasToxyMatrix->MultiplyPoint(handleWorldPos, handleDisplayPos);
     handleDisplayPos[2] = 0.0; // Handles are always projected
   }
   else
   {
-    double* handleWorldPos = handleInfo.PositionWorld;
     this->Renderer->SetWorldPoint(handleWorldPos);
     this->Renderer->WorldToDisplay();
     this->Renderer->GetDisplayPoint(handleDisplayPos);
     handleDisplayPos[2] = 0.0; // Handles are always projected
   }
 
   double dist2 = vtkMath::Distance2BetweenPoints(handleDisplayPos, displayPosition3);
+  if (!interactionEventData->IsDisplayPositionValid())
+  {
+    dist2 = vtkMath::Distance2BetweenPoints(handleWorldPos, interactionEventData->GetWorldPosition());
```

Comment:
Currently, distance is specified in display coordinates (if somewhere this is not respected then it should be fixed).

We cannot simply return distance in world coordinates. We could estimate distance in display coordinates from distance in world coordinates, for example by using the scaling factor at the focal point of the camera (in desktop) or at the controller position (in virtual reality).

We could potentially allow returning of distance in both display and world coordinates and consolidate them at higher level, but that may complicate things a lot.

Reply:
I've removed the code related to "picking" with the controllers in VR. I think that this is going to be a complicated topic that is beyond the scope of this PR.

Reply:
It makes sense. It is better to get this merged and then later implement interaction (grabbing and laser pointer) properly later.

Reply:
> then later implement interaction (grabbing and laser pointer) properly later.

:+1:

Reply:
@jcfr What is the status of laser pointer interaction? Can the laser pointer be used for something already or is it a work in progress? Is someone planned to work on it in the near future?

Reply:
Few pointers from prior notes:
* we have 3D support in [vtkMRMLInteractionEventData](https://github.com/Slicer/Slicer/blob/main/Libs/MRML/DisplayableManager/vtkMRMLInteractionEventData.cxx) and [vtkMRMLViewInteractorStyle::DelegateInteractionEventToDisplayableManagers](https://github.com/Slicer/Slicer/blob/62ec1f5fb15417a2930fcd217a3f7f0cf86eb461/Libs/MRML/DisplayableManager/vtkMRMLViewInteractorStyle.cxx#L295-L327)
* [vtkSlicerROIRepresentation3D::CanInteractWithROI](https://github.com/Slicer/Slicer/blob/62ec1f5fb15417a2930fcd217a3f7f0cf86eb461/Modules/Loadable/Markups/VTKWidgets/vtkSlicerROIRepresentation3D.cxx#L363-L381) does not support 3D
* For reference, in regular VTK handling of 3D is done:
   * [vtkBoxWidget2::SelectAction3D](https://github.com/Kitware/VTK/blob/e0b9ddb4b4cbd83590f82f24e964aa2f9c71384e/Interaction/Widgets/vtkBoxWidget2.cxx#L206-L224)
   * [vtkBoxRepresentation::ComputeComplexInteractionState](https://github.com/Kitware/VTK/blob/e0b9ddb4b4cbd83590f82f24e964aa2f9c71384e/Interaction/Widgets/vtkBoxRepresentation.cxx#L1392-L1459)

Reply:
Does VTK laser picking require that you point somewhere on the handle? Or there is a tolerance, similar to desktop screen interaction (so that you don't need to click exactly on a handle to be able to drag it)?